Code/Framework/AzCore/AzCore/DOM/DomPath.h

```diff
@@ -47,6 +47,11 @@ namespace AZ::Dom
         bool operator!=(const AZ::Name& key) const;
         bool operator!=(AZStd::string_view key) const;
 
+        bool operator<(const PathEntry& other) const;
+        bool operator<(size_t index) const;
+        bool operator<(const AZ::Name& key) const;
+        bool operator<(AZStd::string_view key) const;
```

Comment:
The other functions need free functions equivalents where they are a the first parameter.
It actually would be more consistent to make all these functions friend free functions instead of members.

```suggestion
        friend bool operator<(const PathEntry& entry, size_t index);
        friend bool operator<(size_t index, const PathEntry& entry);
        friend bool operator<(const PathEntry& entry, const AZ::Name& key);
        friend bool operator<(const AZ::Name& key, const PathEntry& entry);
        friend bool operator<(const PathEntry& entry, AZStd::string_view key);
        friend bool operator<(AZStd::string_view key, const PathEntry& entry)
```

Reply:
Swapped the member `operator<` functions for a set of `operator<` friend free functions.